custom_connectors/oauth2/trackvia.rb

```diff
@@ -223,7 +150,7 @@
   object_definitions: {
     record: {
       fields: lambda { |_connection, config_fields|
-        call(:get_fields, config_fields)
+        call(:get_all_output_fields, config_fields)
```

Comment:
@johnhuffmanattrackvia Seems like we're centralising the method for all of them to `get_all_output_fields` but I'm wondering if the filters done in `get_fields` and `get_output_fields` are useful. 

i.e. for `get_fields` we reject `canCreate` and `canUpdate` fields.

Reply:
Also, this makes all fields when creating a record required as well? Would that be correct to say so?

Reply:
I was trying to centralize the logic, but having had some extra time since I opened the PR I'm wondering if its too high risk.  I began that when I realized that the way the input data pills are configured was pretty redundant.  Its also a considerable challenge to make a connector that encapsulates all the permutations of config a low code platform can generate.  Let me back this up a bit and try to distill more of what is needed

Reply:
really what needs to happen is this 
```
get_all_output_fields: lambda do |input|
      url = input[:view_id].present? ? "views/#{input[:view_id]}" : "users"
      Array.wrap(get(url)&.[]('structure')).reject { |field| field['name'] == "ID" }.map do |field|
        case field['type']
        when 'relationship'
          [{ name: "f_#{field['fieldMetaId']}", label: field['name'], field_name: field['name'], type: 'string' },
           { name: "f_#{field['fieldMetaId']}(id)", label: "#{field['name']}(id)", field_name: "#{field['name']}(id)", type: 'number' }]
        when 'number', 'currency', 'percentage', 'autoIncrement',
          { type: 'number' }
        when 'date'
          { type: 'date' }
        when 'datetime'
          { type: 'date_time' }
        when 'point'
          { type: 'object', properties: [
            { name: 'latitude', type: 'number' },
            { name: 'longitude', type: 'number' }
          ] }
        when 'checkbox'
          { type: :array, of: :object, properties: [
            { name: 'choice' }
          ] }
        else
          {}
        end.merge(name: "f_#{field['fieldMetaId']}", label: field['name'], field_name: field['name'])
      end.concat([{ name: 'id' }])
    end
```

In the case of relationship fields, an additional fK is added to the data payload that is not reflected in the structure its looking at.  I was using flatten and compact to turn this into a flat array but using merge on the field meta ID is always going to miss the fK field on the relationship.  thoughts?

Reply:
Hmm, we could separate out relationship handling to outside of the `case when` statement? Since merge isn't playing nice with an array as compared to hashes that come out every other time.

Reply:
@bennettgo how about this instead.  this is a much more concise update and seems to give the result I'm looking for in testing.  This has the smallest impact on the connector and its behavior now